src/Components/CreateImageWizardV2/steps/Packages/Packages.tsx

```diff
@@ -1,545 +1,563 @@
+                onChange={handleRepoToggleClick}
+              />
+              <ToggleGroupItem
+                text="All repos"
+                buttonId="toggle-all-repos"
+                isSelected={toggleSourceRepos === 'toggle-all-repos'}
+                onChange={handleRepoToggleClick}
+              />
+            </ToggleGroup>
+          </ToolbarItem>
+          */}
+          <ToolbarItem variant="pagination">
+            <Pagination
+              itemCount={transformedPackages.length}
+              perPage={perPage}
+              page={page}
+              onSetPage={handleSetPage}
+              onPerPageSelect={handlePerPageSelect}
+              isCompact
+            />
+          </ToolbarItem>
+        </ToolbarContent>
+      </Toolbar>
+      <Table variant="compact" data-testid="packages-table">
+        <Thead>
+          <Tr>
+            <Th />
+            <Th width={20}>Package name</Th>
+            <Th width={35}>Description</Th>
+            <Th width={25}>Package repository</Th>
+            <Th width={20}>Support</Th>
+          </Tr>
+        </Thead>
+        <Tbody>
+          {!searchTerm && toggleSelected === 'toggle-available' && (
+            <EmptySearch />
+          )}
+          {searchTerm && transformedPackages.length === 0 && <NoResultsFound />}
+          {searchTerm &&
+            transformedPackages.length >= 100 &&
+            handleExactMatch()}
+          {transformedPackages.length < 100 &&
+            transformedPackages
+              .sort((a, b) => {
+                const aPkg = a.name.toLowerCase();
+                const bPkg = b.name.toLowerCase();
+                // check exact match first
+                if (aPkg === searchTerm) {
+                  return -1;
+                }
+                if (bPkg === searchTerm) {
+                  return 1;
+                }
+                // check for packages that start with the search term
+                if (
+                  aPkg.startsWith(searchTerm) &&
+                  !bPkg.startsWith(searchTerm)
+                ) {
+                  return -1;
+                }
+                if (
+                  bPkg.startsWith(searchTerm) &&
+                  !aPkg.startsWith(searchTerm)
+                ) {
+                  return 1;
+                }
+                // if both (or neither) start with the search term
+                // sort alphabetically
+                if (aPkg < bPkg) {
+                  return -1;
+                }
+                if (bPkg < aPkg) {
+                  return 1;
+                }
+                return 0;
               })
-            )}
-          </DualListSelectorList>
-        </DualListSelectorPane>
-      </DualListSelector>
+              .slice(computeStart(), computeEnd())
+              .map((pkg, rowIndex) => (
+                <Tr key={`${pkg.name}-${rowIndex}`} data-testid="package-row">
```

Comment:
This is also a pretty big code block and it has the conditional rendering too... could we try to improve readability, maybe by extracting this into a `<PackageRow>` component?

Reply:
Tried to extract the logic to a `<PackageRow>` component, but event though everything renders correctly in msw the tests start to fail, showing only one row in the packages table. or completely sure where the problem is. So I've reverted this change for now. Trying to figure this out.